src/arena.py

```diff
@@ -1 +1,146 @@
-WORKS = True
+import random
+import re
+from dataclasses import dataclass
+from typing import List
+
+
+class Combatant:
+    def __init__(self, name: str, health: int, damage: str):
+        self.name = name
+        self.health = health
+        self.__damage = self.Damage(damage)
+        self.last_damage = 0
+        self.last_health = health
+
+    @property
+    def damage(self) -> str:
+        return self.__damage.to_string()
+
+    def attack(self, other: "Combatant") -> None:
+        self.last_damage = self.__damage.draw()
+        other.__get_attacked(self.last_damage)
```

Comment:
I know that this is the same class, but don't you think the `__get_attacked` method should be public?
We use `other` here as we would use an object of different class with unknown private methods.